pkg/configs/db/db.go

```diff
@@ -7,50 +7,50 @@ import (
 	"io/ioutil"
 	"net/url"
 
-	"github.com/cortexproject/cortex/pkg/configs"
 	"github.com/cortexproject/cortex/pkg/configs/db/memory"
 	"github.com/cortexproject/cortex/pkg/configs/db/postgres"
+	"github.com/cortexproject/cortex/pkg/configs/userconfig"
 )
 
 // Config configures the database.
 type Config struct {
-	URI           string
-	MigrationsDir string
-	PasswordFile  string
+	URI           string `yaml:"uri"`
+	MigrationsDir string `yaml:"migrations_dir"`
+	PasswordFile  string `yaml:"password_file"`
 
 	// Allow injection of mock DBs for unit testing.
 	Mock DB `yaml:"-"`
 }
 
 // RegisterFlags adds the flags required to configure this to the given FlagSet.
 func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
-	f.StringVar(&cfg.URI, "database.uri", "postgres://postgres@configs-db.weave.local/configs?sslmode=disable", "URI where the database can be found (for dev you can use memory://)")
-	f.StringVar(&cfg.MigrationsDir, "database.migrations", "", "Path where the database migration files can be found")
-	f.StringVar(&cfg.PasswordFile, "database.password-file", "", "File containing password (username goes in URI)")
+	f.StringVar(&cfg.URI, "configs.database.uri", "postgres://postgres@configs-db.weave.local/configs?sslmode=disable", "URI where the database can be found (for dev you can use memory://)")
+	f.StringVar(&cfg.MigrationsDir, "configs.database.migrations", "", "Path where the database migration files can be found")
```

Comment:
Given we're introducing breaking changes anyway, I would suggest to rename this flag to `configs.database.migrations-dir` to consistency.